retworkx/custom_return_types.pyi

```diff
@@ -0,0 +1,237 @@
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This file contains only type annotations for PyO3 functions and classes
+# For implementation details, see __init__.py and lib.rs
+
+from typing import Any, Generic, List, Iterable, TypeVar, Tuple
+from collections.abc import Sequence
+
+T = TypeVar("T")
+
+class NodeIndices(Sequence[int]):
+    def __init__(self) -> None: ...
+    def __eq__(self, other) -> bool: ...
+    def __ge__(self, other) -> bool: ...
```

Comment:
The tool I used thinks we have `__ge__` implemented because we have `__richcmp__` implemented, but in fact it just throws an error - so we just ignore it. For `__eq__` we use `Sequence[T]` now